docs/tool-choice.md

```diff
@@ -0,0 +1,348 @@
+# Tool Choice
+
+Tool choice controls which tools the model can use during a request. This is useful for controlling agent behavior, optimizing for specific use cases, or working around provider limitations.
+
+## Overview
+
+PydanticAI distinguishes between two types of tools:
+
+- **Function tools**: Tools you register with the agent using `@agent.tool` or `tools=[...]`
+- **Output tools**: Internal tools used by the framework for structured output (e.g., `final_result`)
+
+The `tool_choice` setting in [`ModelSettings`][pydantic_ai.settings.ModelSettings] controls which **function tools** are available to the model. Output tools are handled separately to ensure agents can always complete their work.
+
+## Options
+
+### `None` or `'auto'` (default)
+
+The model decides whether to use tools. All function tools and output tools are available.
+
+```python {test="skip"}
+from pydantic_ai import Agent
+
+
+def get_weather(city: str) -> str:
+    """Get weather for a city."""
+    return f'Weather in {city}: sunny'
+
+
+def get_time(city: str) -> str:
+    """Get current time for a city."""
+    return f'Current time in {city}: 12:00'
+
+
+agent = Agent('openai:gpt-4o', tools=[get_weather, get_time])
+
+# Both are equivalent - model decides when to use tools
+result = agent.run_sync('What is the weather in Paris?')
+result = agent.run_sync('What is the weather in Paris?', model_settings={'tool_choice': 'auto'})
+```
+
+### `'none'` or `[]`
+
+Disables function tools. The model can only respond with text (or use output tools for structured output).
+
+```python {test="skip"}
+from pydantic_ai import Agent
+
+
+def get_weather(city: str) -> str:
+    """Get weather for a city."""
+    return f'Weather in {city}: sunny'
+
+
+agent = Agent('openai:gpt-4o', tools=[get_weather])
+
+# Function tools disabled - model responds with text only
+result = agent.run_sync(
+    'What is the weather in Paris?',
+    model_settings={'tool_choice': 'none'}
+)
+```
+
+This is useful when you want to prevent tool calls before hitting usage limits or when you want the model to respond directly without using tools.
+
+!!! note "Output tools remain available"
+    When using structured output, output tools are still available even with `tool_choice='none'`. This ensures agents can complete with the required output type.
+
+### `'required'`
+
+Forces the model to use a function tool. No output tools are sent.
```

Comment:
This implies we don't send output tools and always break the cache, which is not true. I'd rather have a separate paragraph about how this interacts with output tools, like we have above